pkg/controller/s3/s3bucket.go

```diff
@@ -183,6 +186,28 @@ func (r *Reconciler) _sync(bucket *bucketv1alpha3.S3Bucket, client s3.Service) (
 	return result, r.Update(ctx, bucket)
 }
 
+// helper function to perform S3 Bucket Versions update
+func updateVersioning(bucket *bucketv1alpha3.S3Bucket, bucketInfo *s3.Bucket, client s3.Service) error {
+	if bucketInfo.Versioning != bucket.Spec.Versioning {
+		err := client.UpdateVersioning(bucket)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// helper function to perform S3 Bucket Tags update
+func updateTagging(bucket *bucketv1alpha3.S3Bucket, client s3.Service) error {
+	if bucket.Spec.Tags != nil {
```

Comment:
I see that we're not checking whether we need an update or not. Is this because `PutBucketTagging` is idempotent?

Reply:
Would probably be good to still not make the call even if idempotent

Reply:
Yes it should be idempotent, the condition is just to cater CR that don't required tagging so it doesn't make unnecessary API call

Reply:
I think what @hasheddan means is that the call is being made every time as long as there are tags on the resource and that may not be good even if it doesn't return an error. One reason is that unnecessary API calls could fill the API rate limit of AWS and another reason is that some resources get in an `updating` state even if the update call doesn't change anything. I think the latter does not apply here since tags are considered as separate resource in AWS and do not affect the state of the main resouce, the former argument still holds but to prevent that we'd need to make a Get call to check the change. @hasheddan do you think it's worth adding that call given that this code piece will be replaced by v1beta1 by the release anyway?

Reply:
I think it is fine to skip for now :)